beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -340,10 +336,8 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
                         // - Poisoned -> this is an intermediate state that should never be reached
                         // - AwaitingDownload -> A recoverable failed batch should have been
                         //   re-requested.
-                        unreachable!(
-                            "Optimistic batch indicates inconsistent chain state: {:?}",
-                            state
-                        )
+                        debug!(self.log, "Optimistic batch indicates inconsistent chain state"; "state" => ?state);
```

Comment:
should we make this a crit if its supposed to be unreachable?

Reply:
yep